src/controllers/movies/movies.js

```diff
@@ -30,11 +30,11 @@ export default function (view, params, tabContent, options) {
     }
 
     function shuffle() {
-        ApiClient.getItem(
-            ApiClient.getCurrentUserId(),
-            params.topParentId
-        ).then((item) => {
-            playbackManager.shuffle(item);
+        isLoading = true;
+        loading.show();
+        const newQuery = { ...query, SortBy: 'Random', Limit: 1 };
+        return ApiClient.getItems(ApiClient.getCurrentUserId(), newQuery).then(({ Items }) => {
+            playbackManager.shuffle(Items[0]);
         });
```

Comment:
_I haven't tested - reviewing from mobile._
Shouldn't we reset `isLoading`? Otherwise, I believe the prev/next buttons will be blocked.
```suggestion
        }).finally(() => {
            isLoading = false;
        });
```
Logically, we should call `loading.hide()` as well, but this might hide the one from PlaybackManager.

Reply:
I copied the pattern from the function right above this one. 

```
function fetchData() {
        isLoading = true;
        loading.show();
        return ApiClient.getItems(ApiClient.getCurrentUserId(), query);
    }
```


The changes I made here to set the loading flag and show the loading element aren't necessary to fix the filtered shuffle feature. It was just a "drive by" improvement, as it is nice to see some immediate indication that the shuffle button was clicked.

I can remove these unrelated changes if the PR is better that way.

Reply:
`fetchData` is used to refresh the item list, and `isLoading = true` - to block intermediate clicking on prev/next buttons.
Then `afterRefresh` is called and resets `isLoading`: https://github.com/jellyfin/jellyfin-web/blob/a419aa9e89b6991fe4fc652abe2fafba166998bf/src/controllers/movies/movies.js#L91

Just try:
1. Shuffle movies.
2. Back from the playback.
3. Click prev/next button.

Reply:
Confirmed that this would introduce a bug with the prev/next buttons.  I'll make your suggested changes soon.